animation_nodes/nodes/number/number_range.py

```diff
@@ -57,13 +62,20 @@ def getExecutionFunctionName(self):
             if self.floatStepType == "START_STEP":
                 return "execute_FloatRange_StartStep"
             elif self.floatStepType == "START_STOP":
-                return "execute_FloatRange_StartStop"
+                if self.endPoint:
+                    return "execute_FloatRange_StartStop_EndPoint"
+                else:
+                    return "execute_FloatRange_StartStop"
 
     def execute_IntegerRange(self, amount, start, step):
         return range_LongList_StartStep(amount, start, step)
 
     def execute_FloatRange_StartStep(self, amount, start, step):
         return range_DoubleList_StartStep(amount, start, step)
 
-    def execute_FloatRange_StartStop(self, amount, start, stop):
+    def execute_FloatRange_StartStop_EndPoint(self, amount, start, stop):
         return range_DoubleList_StartStop(amount, start, stop)
+
+    def execute_FloatRange_StartStop(self, amount, start, stop):
+        return range_DoubleList_StartStep(amount, start,
```

Comment:
Better use a guard condition to check for `amount <= 0` here. That makes the expression below a bit simpler.